src/odemis/acq/stream/_sync.py

```diff
@@ -1591,17 +1597,18 @@ def _runAcquisitionScanStage(self, future):
                                           "acquisitions.")
 
         sstage = self._sstage
+        if not sstage:
+            raise ValueError("Cannot acquire with scan stage, as no stage was provided")
+        orig_spos = sstage.position.value  # TODO: need to protect from the stage being outside of the axes range?
+        keep_stage_pos = model.getComponent(role="stage").name in sstage.affects.value
```

Comment:
maybe rename to `scan_stage_is_stage` (or something like that). This way, you can use this condition also line 1752.